### PR TITLE
Add test leg to run runtime tests on Android arm64

### PR DESCRIPTION
**Pull request by fanyang-mono**
This work contributes to the project of bringing up test coverage for Mono (#33381)

488 runtime tests ran and passed on this newly added CI lane. 1856 runtime tests were disabled for this lane, due to long running work items and oversized Android apps. Created two issues to track those two issues separately. (#45965 and #45568)

### CODE REVIEW

**Comment by SamMonoRT** [comment]
@safern - this is a new lane too, what needs to be done to add it to the staging pipeline first ?

**Comment by naricc** [comment]
@SamMonoRT So, my understanding is basically that to add to staging, you add the lanes to runtime-staging.yml instead of runtime.yml. You can use @safern's original PR moving the android emulator legs: https://github.com/dotnet/runtime/pull/45178/files

**Review by trylek — APPROVED** [review]
Looks good, thank you! Please make sure to revert the instrumentation changes to runtime.yml before merging in :-).

**Review by safern — COMMENTED** [review]
Let's add it to `runtime-staging` pipeline first to make sure it is stable before we move it to the main pipeline.

**Review by naricc — APPROVED** [review]
Let's put it in staging first as @SamMonoRT suggested. Other than that, looks good.

**Comment by fanyang-mono** [comment]
According to the log [here](https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-44560-merge-9a8c3c1761ca4af88e/JIT.Regression/console.3021f45f.log?sv=2019-07-07&se=2020-12-29T19%3A38%3A02Z&sr=c&sp=rl&sig=BkHcaP95O8Xj8%2Bzd6aUgi0sWv9Kwivyv92kazQGF91A%3D). On some Android devices, it took ~2.5min to install an Android app (size: 11.4 Mb). This work item contains one test wrapper, which includes 82 tests. That means this work item does 82 times app installation, which is more than 3 hours. Some devices are faster than the others. It seems to me that the devices were picked randomly for each work item. 

So the best solution is to avoid unnecessary installations. For each test wrapper, all tests use the same Android app. @akoeplinger suggested to install/uninstall once in the initialize/cleanup methods for each test wrapper. That would require this functionality(https://github.com/dotnet/xharness/issues/397) in place. So for this PR, I will probably have to disable quite a few tests. And enable them when that functionality is ready to use.

**Comment by fanyang-mono** [comment]
This issue contributes to the project of bringing up test coverage for Mono (https://github.com/dotnet/runtime/issues/33381)

**Review by akoeplinger — APPROVED** [review]

File: src/tests/Directory.Build.props [hunk 1]
```diff
@@ -135,4 +135,11 @@
     <NuGetTargetMoniker>$(NetCoreAppCurrentToolTargetFrameworkMoniker)</NuGetTargetMoniker>
     <NuGetTargetMonikerShort>$(NetCoreAppToolCurrent)</NuGetTargetMonikerShort>
   </PropertyGroup>
+
+  <!-- Set Test Wrapper running host OS -->
+  <PropertyGroup>
+    <TestWrapperTargetsWindows>false</TestWrapperTargetsWindows>
+    <TestWrapperTargetsWindows Condition=" ('$(TargetsWindows)' != '' And '$(TargetsWindows)' ) OR ('$(TargetOS)' == 'Android' And '$(TargetArchitecture)' == 'arm64' )">true</TestWrapperTargetsWindows>
```
Comment (sandreenko) [inline src/tests/Directory.Build.props:142]: Do we use ".cmd" scripts on arm64 android?

Comment (trylek) [reply src/tests/Directory.Build.props:142]: I think the trick is that, while traditional CoreCLR builds deal with a pair of platforms (hosting / targeting), for Mono device testing there's a triplet of platforms - host, test driver, target. Arm64 Android tablets use Windows test drivers so at that level yes, they use ".cmd" scripts.

**Pull request merged**